src/v1/providers/database.ts

```diff
@@ -598,6 +612,11 @@ export class DataSnapshot {
       }
     }
 
+    if (numKeys === 0) {
+      // Empty node
+      return null;
+    }
+
```

Comment:
This simulates the `this.isEmpty()` [check the Firebase SDK does](https://github.com/firebase/firebase-js-sdk/blob/8599d91416ae8ac5202742f11cee00666d3360ec/packages/database/src/core/snap/ChildrenNode.ts#L198-L200).